### PR TITLE
HTCONDOR-692:  Utilize user-provided hostname, not canonical name, for alias.

### PR DESCRIPTION
**Pull request by Todd-L-Miller**
See #572; this PR just adds the sugggested release note and corrects a typo in the original commit message.

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [x] Check for correctness of change
- [x] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [x] Check for documentation, if needed
- [x] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

**Comment by Todd-L-Miller** [comment]
From #572:

> I can't speak to the security implications of the change, but the code is fine and the change seems sane.
>
> This could break currently-functioning SSL set-ups if the cert in use includes only the ANAME and not the CNAME (as a
> SAN). I'll ask ToddT about adding a hidden knob that effectively reverts the change at run-time.

**Comment by Todd-L-Miller** [comment]
ToddT indicated that there should be a knob, and that it should default to the new behavior.

File: src/condor_daemon_client/daemon.cpp [hunk 1]
```diff
@@ -1584,7 +1584,7 @@ Daemon::findCmDaemon( const char* cm_name )
 			return false;
 		}
 		sinful.setHost(saddr.to_ip_string().c_str());
-		sinful.setAlias(fqdn.c_str());
+		sinful.setAlias(host);
```
Comment (JaimeFrey) [inline src/condor_daemon_client/daemon.cpp:1587]:
```suggestion
```
Deleting this call matches the logic in getDaemonInfo() when hostname:port is given. The call to New_alias() here should cause the given alias to be added to the sinful that's set in New_addr().

**Review by JaimeFrey — CHANGES_REQUESTED** [review]

**Review by JaimeFrey — APPROVED** [review]
Looks good now.

**Pull request merged**